addons/mass_mailing/tests/test_mailing_internals.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mail.tests.common import TestMail
+from odoo.addons.mass_mailing.data.mail_test_data import MAIL_TEMPLATE
+
+
+class TestMailingInternals(TestMail):
+
+    def setUp(self):
+        super(TestMailingInternals, self).setUp()
+
+        self.bounce_alias = 'bounce.test'
+        self.catchall_alias = 'catchall.test'
+        self.catchall_domain = 'test.example.com'
+        self.env['ir.config_parameter'].set_param('mail.bounce.alias', self.bounce_alias)
+        self.env['ir.config_parameter'].set_param('mail.catchall.alias', self.catchall_alias)
+        self.env['ir.config_parameter'].set_param('mail.catchall.domain', self.catchall_domain)
+
+        self.test_alias = self.env['mail.alias'].create({
+            'alias_name': 'test.alias',
+            'alias_user_id': False,
+            'alias_model_id': self.env['ir.model']._get('mail.test.simple').id,
+            'alias_contact': 'everyone'
+        })
+
+    def test_mailing_trace_update(self):
+        customers = self.env['res.partner']
+        for x in range(0, 3):
+            customers |= self.env['res.partner'].create({
+                'name': 'Customer_%02d' % x,
+                'email': '"Customer_%02d" <customer_%02d@test.example.com' % (x, x),
+            })
```

Comment:
Could use a create on a list comprehension instead?
Not a fan of create calls in loops (even if this is a tiny one).

Reply:
Well it allows to make a record set. And I am not sure create in batch on partner is allowed in 11 ...

Reply:
```py
customers = self.env['res.partner'].create([{
    'name': 'Customer_%02d' % x,
     'email': '"Customer_%02d" <customer_%02d@test.example.com' % (x, x),
} for x in range(0, 3)])
```

Would also make you a nice recordset no?
If it's not supported it's whatever I guess, but even the construction is a bit nicer IMO.

Reply:
In 11 create is not multi-enabled on partner model (still an api model).